### PR TITLE
Add dynamic API_VERSION and BUILD_NUMBER counters and minor codestyle changes

### PR DESCRIPTION
**Pull request by adydychk**

### CODE REVIEW

**Comment by adydychk** [comment]
In first version functions added directly to the configurations.
`get_api_version(repo_path)` was rewritten to search line by line

**Comment by adydychk** [comment]
@azhogov What is better to return in case of nothing found in `get_api_version(repo_path)`?
Or better to fail?

File: conf_linux_public.py [hunk 1]
```diff
@@ -24,12 +24,12 @@
 import git
 
 #TODO: move functions to the shared module
-def get_commit_number(repo_path):
+def set_commit_number(repo_path):
     git_repo = git.Git(repo_path)
     commits = git_repo.log('--all', '--oneline')
-    return len(commits.splitlines())
+    DEFAULT_OPTIONS["BUILD_NUMBER"] = len(commits.splitlines())
```
Comment (azhogov) [inline conf_linux_public.py]: will not work :)

File: conf_linux_public.py [hunk 2]
```diff
@@ -59,8 +59,14 @@ def get_api_version(repo_path):
 
 MEDIA_SDK_REPO_DIR = DEFAULT_OPTIONS.get('REPOS_DIR') / PRODUCT_REPOS[0]['name']
 
-BUILD_NUMBER = get_commit_number(str(MEDIA_SDK_REPO_DIR))
-API_VERSION = get_api_version(MEDIA_SDK_REPO_DIR)
+action('count build_number',
+       callfunc=(set_commit_number, [str(MEDIA_SDK_REPO_DIR)], {}))
```
Comment (azhogov) [inline conf_linux_public.py]: probably set_build_number

File: conf_linux_public_api_latest.py [hunk 3]
```diff
@@ -40,14 +40,14 @@ def get_api_version(repo_path):
...
-    file_path = pathlib.Path(repo_path) / 'api' / 'include' / 'mfxdefs.h'
+    mediasdk_api_header_path = pathlib.Path(repo_path) / 'api' / 'include' / 'mfxdefs.h'
 
-    with open(file_path, 'r') as file:
-        for line in file:
-            if re.search("MFX_VERSION_MAJOR\s(\d+)", line):
-                major_version = re.search("MFX_VERSION_MAJOR\s(\d+)", line).group(1)
-                minor_version = re.search("MFX_VERSION_MINOR\s(\d+)", next(file)).group(1)
-                return f"{major_version}.{minor_version}"
+    with open(mediasdk_api_header_path, 'r') as lines:
```
Comment (azhogov) [inline conf_linux_public_api_latest.py]: "as api_file_obj" is better

File: conf_linux_public_api_latest.py [hunk 4]
```diff
@@ -40,14 +40,14 @@ def get_api_version(repo_path):
...
+        for line in lines:
+            major_version = re.search("MFX_VERSION_MAJOR\s(\d+)", line)
+            if major_version:
+                minor_version = re.search("MFX_VERSION_MINOR\s(\d+)", next(lines))
+                DEFAULT_OPTIONS["API_VERSION"] = f"{major_version.group(1)}.{minor_version.group(1)}"
```
Comment (azhogov) [inline conf_linux_public_api_latest.py]: need to check minor_version before using :)

File: conf_linux_public.py [hunk 5]
```diff
@@ -18,18 +18,15 @@
...
 #TODO: move functions to the shared module
 def set_commit_number(repo_path):
+    import git
     git_repo = git.Git(repo_path)
     commits = git_repo.log('--all', '--oneline')
-    DEFAULT_OPTIONS["BUILD_NUMBER"] = len(commits.splitlines())
+    DEFAULT_OPTIONS["ENV"]["BUILD_NUMBER"] = len(commits.splitlines())
 
-def set_api_version(repo_path):
+def set_additional_env(repo_path):
```
Comment (azhogov) [inline conf_linux_public.py]: set_build_env

**Pull request merged**